packages/postgraphile-core/__tests__/kitchen-sink-schema.sql

```diff
@@ -1238,3 +1239,125 @@ create table geometry.geom (
+
+create function function_returning_enum.applicants_next_stage(
+  a function_returning_enum.applicants
+) returns function_returning_enum.stage_options_enum_domain
+as $$
+  select (case when a.stage = 'round 2' then 'hired' 
+    else 'rejected' end)::function_returning_enum.stage_options_enum_domain;
+$$ language sql stable;
+comment on function function_returning_enum.applicants_next_stage is E'@filterable';
+
+create table function_returning_enum.length_status (
+  type text primary key
+);
+comment on table function_returning_enum.length_status is E'@enum';
+insert into function_returning_enum.length_status (type) values ('ok'), ('too_short');
+
+--- use an @enum tag this time
+create domain function_returning_enum.length as text;
+comment on domain function_returning_enum.length is E'@enum length_status';
+
+create function function_returning_enum.text_length(
+  text text,
+  min_length int
+) returns function_returning_enum.length
+as $$
+  select (case when length(text) < min_length then 'too_short' 
+    else 'ok' end)::function_returning_enum.length;
+$$ language sql stable;
+
+create function function_returning_enum.applicants_name_length(
+  a function_returning_enum.applicants
+) returns function_returning_enum.length
+as $$
+  select function_returning_enum.text_length(a.last_name, 4)::function_returning_enum.length;
+$$ language sql stable;
+comment on function function_returning_enum.applicants_name_length is E'@filterable';
+
+create function function_returning_enum.applicants_by_stage(
+  wanted_stage function_returning_enum.stage_options_enum_domain
+) returns setof function_returning_enum.applicants
+as $$ 
+  select * from function_returning_enum.applicants a where a.stage = wanted_stage;
+$$ language sql stable;
+
+create function function_returning_enum.applicants_by_favorite_pet(
+  pet function_returning_enum.animal_type
+) returns setof function_returning_enum.applicants
+as $$ 
+  select * from function_returning_enum.applicants a where a.favorite_pet = pet;
+$$ language sql stable;
+
+create function function_returning_enum.applicants_pet_food(
+  a function_returning_enum.applicants
+) returns function_returning_enum.animal_type
+as $$
+  select (case 
+    when a.favorite_pet = 'FISH' then null 
+    when a.favorite_pet = 'CAT' then 'FISH'
+    when a.favorite_pet = 'DOG' then 'CAT'
+    else null
+    end)::function_returning_enum.animal_type;
+$$ language sql stable;
+comment on function function_returning_enum.applicants_pet_food is E'@filterable';
+
+create domain function_returning_enum.transportation as text;
+comment on domain function_returning_enum.transportation is E'@enum enum_table_transportation_enum';
+
+create function function_returning_enum.applicants_by_transportation(
+  transportation function_returning_enum.transportation
+) returns setof function_returning_enum.applicants
+as $$ 
+  select * from function_returning_enum.applicants a where a.transportation = transportation;
```

Comment:
Took me a while to figure out why `applicantsByTransportation` returned all three records!

```suggestion
  select * from function_returning_enum.applicants a where a.transportation = applicants_by_transportation.transportation;
```

Reply:
arf I get bitten by this so often, damn ambiguous names in postgres!